sdk/repo/verify.go

```diff
@@ -232,10 +264,32 @@ func VerifySync(name string) error {
 			}
 
 			if !found {
-				plog.Errorf("Project dir %s at %q, expected %s",
+				plog.Errorf("Project dir %s at branches %q, expected %s (ignore the refs/heads/ prefix)",
 					project.Path, branches, project.Revision)
 				result = VerifyError
 			}
+		} else if strings.HasPrefix(project.Revision, "refs/tags/") {
+			tags, err := manifest.projectTags(project)
+			if err != nil {
+				return err
+			}
+
+			found := false
+
+			for _, tag := range tags {
```

Comment:
This block wouldn't be needed if the result from the function was a map of bool rather than a list. Given that you're creating it specially for this, wouldn't it make more sense to return a map?

Reply:
Yes, or Go could have a function to check if a slice contains a value, instead of everybody rewriting it, let's wait for generics?

Reply:
Empty structs or leave this for loop?

Reply:
I don't care about empty structs vs boolean. I personally find the boolean clearer, but it might just be because I've been using that "set" implementation for a while.  Whatever you think is clearer.

I really prefer to avoid a useless for-loop, not because of optimization, but rather because the code is just harder to read with that for-loop in the middle.

Reply:
Rewrote it with maps but that needed the `reflect.ValueOf(tags).MapKeys()` trick to have a list of found tags to print out.